### PR TITLE
ember-routing test cleanup

### PR DESCRIPTION
**Pull request by seanpdoyle**
- Use test utils where applicable
- reduce variable scope
- don't access `_` prefixed properties when unnecessarily

### CODE REVIEW

File: packages/ember-routing/tests/system/router_test.js [hunk 1]
```diff
@@ -25,42 +26,38 @@ QUnit.module("Ember Router", {
...
 
-  run(container, 'destroy');
+  runDestroy(container);
 
   ok(location.isDestroyed, "location should be destroyed");
 });
 
 test("should instantiate its location with its `rootURL`", function() {
-  createRouter({
+  var router = createRouterWithContainer({
```
Comment (stefanpenner) [inline packages/ember-routing/tests/system/router_test.js]: i think we can leave this named as `createRouter` and just assume with  a container, unless container:null is passed

File: packages/ember-routing/tests/system/router_test.js [hunk 2]
```diff
@@ -25,42 +26,38 @@ QUnit.module("Ember Router", {
...
 });
 
 test("should destroy its location upon destroying the routers container.", function() {
-  createRouter();
-
+  var router = createRouterWithContainer();
   var location = router.get('location');
 
-  run(container, 'destroy');
+  runDestroy(container);
```
Comment (stefanpenner) [inline packages/ember-routing/tests/system/router_test.js]: thoughts on `runAndDestroy`?

Comment (seanpdoyle) [inline packages/ember-routing/tests/system/router_test.js]: How about `seekAndDestroy`?

I don't think the `And` is necessary. The [helper](https://github.com/emberjs/ember.js/blob/e53f5709d05a872447c039048acaef03c0a3bfc6/packages/ember-runtime/tests/utils.js#L7-L11) was extracted by a popular pattern, and merely puts a name to

```js
run(function(foo) { foo.destroy(); });
```

Comment (stefanpenner) [inline packages/ember-routing/tests/system/router_test.js]: runDestroy seems fine.

**Pull request merged**